### ISSUE TITLE
Add Bianchi overborrowing model

### ISSUE THREAD

**Issue by jstac** [QuantEcon/cbc_workshops#3]
Add central planner and decentralized equilibrium code to at least replicate Fig 1 of http://www.javierbianchi.com/uploads/8/8/5/8/8858198/overborrowing.pdf

I am closing https://github.com/QuantEcon/cbc_workshops/issues/2 and https://github.com/QuantEcon/lecture-python-advanced.myst/issues/121 in favor of this issue.

**Comment by jstac** [comment]
Note that the code supplied by @crondonm is addressed to a slightly different model, but I think we can (a) modify it slightly for the task above.  Does that sound right to you @crondonm?  

If not, we can use an Euler equation approach, following the Matlab code.

The current code set is here: https://github.com/QuantEcon/cbc_workshops/tree/main/bianchi

I've converged ipynb files to py files so that it's easier to version control and discuss them.  (Those who prefer ipynb files can convert them backward and forward with jupytext.)

**Comment by jstac** [comment]
Many thanks to @Flintro and @crondonm for contributing code.  To replicate Fig 1, we still need to get suitable output for the planner problem.

**Comment by crondonm** [comment]
Hi @jstac,

Under ref_material, I've added references to a paper by Martín Uribe and Stephanie Smith-Grohé to compute the model for an SOE with downward nominal wage rigidities. 

My code for the planner ([carlos_planner_jax.py](https://github.com/QuantEcon/cbc_workshops/blob/main/bianchi/python/carlos_code/carlos_planner_jax.py)) is based on the "optimal exchange rate" model included in the paper.

The main difference with Bianchi is that the occasionally binding constraint is over the wage instead of the debt level. In particular, the model requires the following:

$$
W_t >=  \gamma W_{t-1}
$$

where $W_t$ is the nominal wage and $\gamma > 0$. However, in this case, the planner always chooses $w_t$ (the real salary) such that the labor market is always at full capacity and the constraint in nominal values is always satisfied with equality. 

In other words, for the referenced code, the constraint becomes irrelevant (for computational purposes).

I will work over the week to get a working example of the Bianchi planner.

**Comment by shlff** [comment]
> Many thanks to @Flintro and @crondonm for contributing code. To replicate Fig 1, we still need to get suitable output for the planner problem.

Hi @jstac I think I've fixed the code issue in the social planner problem and now the code can properly replicate Figure 1 of 
- http://www.javierbianchi.com/uploads/8/8/5/8/8858198/overborrowing.pdf

Please find my code shared via GH gist here:
- https://gist.github.com/shlff/b0fab232cec75165537004135cc00fe4

(For the replication of Figure 1 please refer to code cell 32 in the above gist link.)

Looking forward to your comments.
